apps/ui/pages/reservation-unit/[id].tsx

```diff
@@ -377,54 +413,143 @@ const ReservationUnit = ({
   isPostLogin,
   mapboxToken,
   apiBaseUrl,
+  searchDuration,
+  searchDate,
+  searchTime,
 }: PropsNarrowed): JSX.Element | null => {
   const { t, i18n } = useTranslation();
   const router = useRouter();
   const now = useMemo(() => new Date(), []);
   const isMobile = useMedia(`(max-width: ${breakpoints.m})`, false);
 
-  const [focusDate, setFocusDate] = useState(new Date());
   const [calendarViewType, setCalendarViewType] = useState<WeekOptions>("week");
-  const [initialReservation, setInitialReservation] =
-    useState<PendingReservation | null>(null);
-  const [isReserving, setIsReserving] = useState(false);
   const [isDialogOpen, setIsDialogOpen] = useState(false);
   const [errorMsg, setErrorMsg] = useState<string | null>(null);
-  const [shouldUnselect, setShouldUnselect] = useState(0);
-  const [storedReservation, , removeStoredReservation] =
-    useLocalStorage<PendingReservation>("reservation");
 
   const calendarRef = useRef<HTMLDivElement>(null);
-  const openPricingTermsRef = useRef<HTMLAnchorElement>(null);
+
   const hash = router.asPath.split("#")[1];
 
-  const isClientATouchDevice = isTouchDevice();
+  const reservableTimeSpans = useMemo(
+    () => filterNonNullable(reservationUnit?.reservableTimeSpans),
+    [reservationUnit?.reservableTimeSpans]
+  );
+  const todaysTimeSpans = reservableTimeSpans.filter(
+    (span) => span.startDatetime && isToday(new Date(span.startDatetime))
+  );
```

Comment:
Why `useMemo`? the second filter is slower than the one inside `useMemo`

Reply:
Ahhh - that's a brainfart. My thinking was that it should be on `todaysTimeSpans` (..so that we only filter today's time spans when `reservableTimeSpans` has changed).